hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -943,14 +943,18 @@ public static boolean scmInit(OzoneConfiguration conf,
       }
     } else {
       clusterId = scmStorageConfig.getClusterID();
+      if (!scmStorageConfig.checkPrimarySCMIdInitialized()) {
```

Comment:
I am sorry for now we should setPrimaryScmNodeId only when HA, otherwise it will affect upgraded secure cluster.
As we assume that if primaryScmNodeId is set it means it is the cluster which has done init with new code.

Reply:
This will be problem on clusters init runs every time during SCM start.

Reply:
Done